### PR TITLE
server close connections actively

### PR DESCRIPTION
**Pull request by icodinglife**
server close connections actively

### CODE REVIEW

**Comment by carllerche** [comment]
Would you be able to add comments explaining the changes? The goal is to document, via comments, all the logic.

**Comment by icodinglife** [comment]
ok, thanks for review, i have add comments on it.

File: src/server.rs [hunk 1]
```diff
@@ -187,10 +187,13 @@ pub async fn run(listener: TcpListener, shutdown: impl Future) -> crate::Result<
     let Listener {
         mut shutdown_complete_rx,
         shutdown_complete_tx,
+        notify_shutdown,
         ..
     } = server;
 
     drop(shutdown_complete_tx);
+    // drop `notify_shutdown`, so that the `Handler` can get shutdown notify to close the clients socket.
```
Comment (LucioFranco) [inline src/server.rs:195]:
```suggestion
    // Drop `notify_shutdown`, so that the `Handler` can get shutdown notify to close the clients socket.
```

**Comment by carllerche** [comment]
Hmm... CI hung?

Is CI configured incorrectly?

**Comment by Darksonn** [comment]
I don't know what happened to CI, but this has since been fixed in  #45.

**Pull request closed without merging**